lib/src/Base/Stat/SampleImplementation.cxx

```diff
@@ -1573,6 +1573,32 @@ void SampleImplementation::sortUniqueInPlace()
   if (last + 1 < size_) erase(last + 1, size_);
 }
 
+Indices SampleImplementation::argsort(Bool isIncreasing) const
+{
+  Collection< std::pair<Point, UnsignedInteger> > pointsPairs(size_);
+  for (UnsignedInteger i = 0; i < size_; ++i)
+  {
+    if (isIncreasing)
+    {
+      pointsPairs[i] = std::pair<Point, UnsignedInteger>((*this)[i], i);
+    }
+    else
+    {
+      const Point pointI(static_cast<Point>((*this)[i]));
+      const Point zero(dimension_);
+      const Point oppositePoint(zero - pointI);
+      pointsPairs[i] = std::pair<Point, UnsignedInteger>(oppositePoint, i);
```

Comment:
And something like
```
pointsPairs[i] = std::pair<Point, UnsignedInteger>(pointI * (-1.0), i);
```
This way, no need for two additional points. You can also provide an alternate comparison operator for this case.

Reply:
I created a unitary minus operator for Points. Fixed.